src/main/java/com/epam/session3/Session3Processor.java

```diff
@@ -0,0 +1,118 @@
+        this.executorService = Executors.newFixedThreadPool(sizeOfThreadPool);
+
+    }
+
+    public void startProcess() throws Exception{
+        startWriters();
+        startEnrichers();
+        startReaders();
+    }
+
+    private Collection<String> detectFiles(){
+        String[] files = new File(this.dataDir).list(new FilenameFilter() {
+            @Override
+            public boolean accept(File dir, String name) {
+                return name.matches("DfTRoadSafety_Accidents_[0-9]{4}\\.csv");
+            }
+        });
+        return Arrays.asList(files);
+    }
+
+    private void startWriters() throws IOException {
+        executorService.execute(new AccidentBatchWriter(daytimeOutgoingQueue, dataDir+"/DaytimeAccidents.csv"));
+        executorService.execute(new AccidentBatchWriter(nighttimeOutgoingQueue, dataDir+"/NighttimeAccidents.csv"));
+    }
+
+    private void startEnrichers(){
+        for(int i=0;i<sizeOfEnricherThreads;i++){
+            executorService.execute(new AccidentBatchEnricher(policeForceService, incomingQueue, daytimeOutgoingQueue, nighttimeOutgoingQueue));
+        }
+    }
+
+    private void startReaders(){
+        for(int i=0;i<sizeOfReaderThreads;i++){
+            new Thread(
+                    new Runnable() {
+                        @Override
+                        public void run() {
+                            try {
+                                while (!filenameQueue.isEmpty()) {
+                                    String filename = filenameQueue.poll();
+                                    System.out.println("Start reading " + filename);
+                                    Future<Integer> readerTask = executorService.submit(new AccidentBatchLoader(batchSize, incomingQueue, dataDir + "/" + filename));
+                                    Integer counter = readerTask.get();
+                                    System.out.println("Done of reading " + filename +" counter is " + counter);
+                                }
+                            } catch (Exception e) {
+                                e.printStackTrace();
+                            }
+                        }
+                    }
+            ).start();
+        }
+    }
+
+    public static void main(String[] args) throws Exception{
+        String dataDir = "src/main/resources";
+        int sizeOfIncomingQueue = 2;
+        int sizeOfOutgoingQueue = 2;
+        int sizeOfThreadPool = 20;
+        int sizeOfReaderThreads = 3;
+        int sizeOfEnricherThreads = 5;
+        int sizeOfBatch = 40000;
+        Session3Processor processor = new Session3Processor(
+                sizeOfIncomingQueue,
+                sizeOfOutgoingQueue,
+                sizeOfThreadPool,
+                sizeOfReaderThreads,
+                sizeOfEnricherThreads,
+                sizeOfBatch,
+                dataDir);
+        processor.startProcess();
+    }
```

Comment:
Thanks for you reviewing.
I have done some refactoring according to your comments.